docs/contracts/v2/guides/staking/02-stake.md

```diff
@@ -0,0 +1,71 @@
+---
+id: "stake"
+sidebar_position: 2
+title: "Stake"
+---
+
+The `stake` function takes `tokenId` as input, and stake the stream by transferring the Sablier NFT from the user's
+wallet to the staking contract.
+
+```solidity
+function stake(uint256 tokenId) external {
+    // code goes here.
+}
+```
+
+First step would be to check if the underlying token vesting through the stream is same as the reward token. Since
+Sablier protocol can be used to stream any token, this ensures that staking contract does not accept streams that are
+not vesting the token specified.
+
+```solidity
+if (sablierLockup.getAsset(tokenId) != rewardERC20Token) {
+    revert DifferentStreamingAsset(tokenId, rewardERC20Token);
+}
+```
+
+As mentioned in the assumptions, the contract only allows staking one NFT at a time. So now we will check if the user
+has already staked or not.
+
+```solidity
+if (stakedTokenId[msg.sender] != 0) {
+    revert AlreadyStaking(msg.sender, stakedTokenId[msg.sender]);
+}
+```
+
+Finally, we will set some storage variables and transfer the NFT:
+
+```solidity
+// Effect: store the `msg.sender` as the staker of the token ID.
+stakedAssets[tokenId] = msg.sender;
+
+// Effect: Store the new tokenId against the `msg.sender`.
```

Comment:
The comments are superfluous here because we have the explanations in the docs. We can remove them to lower the maintenance cost.